### PR TITLE
IGAPP-778: Enable noUncheckedIndexedAccess flag and add typescript parser rules

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

This is the last linting PR, promised! Or at least, the last linting PR for integreat-app :stuck_out_tongue: 

I was not sure about enabling the typescript flag [noUncheckedAccessIndex](https://www.typescriptlang.org/tsconfig#noUncheckedIndexedAccess) since there were quite some false positive errors, but I think it really reduces possible error sources since one is forced to always check array elements or map values if they really exist. Let me know what you think!

### CODE REVIEW

File: build-configs/tsconfig.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "extends": "../tsconfig.base.json",
+  "extends": "../tsconfig.json",
```
Comment (maxammann) [inline build-configs/tsconfig.json:2]: Why this change?

File: web/src/components/Tooltip.tsx [hunk 2]
```diff
@@ -208,18 +208,11 @@ const fixFlow = (element: Element | null, preferredFlow: FlowType, dimensions: V
...
-  }
-
   if (checker.check(element.getBoundingClientRect(), dimensions)) {
     return preferredFlow
   }
   const fallback = checker.fallbacks.find((fallbackFlow: FlowType) => {
     const fallbackChecker = spaceCheckers[fallbackFlow]
-    if (!fallbackChecker) {
-      throw new Error('Fallback not found')
-    }
```
Comment (maxammann) [inline web/src/components/Tooltip.tsx:222]: These checks should stay.

File: web/tools/webpack.config.ts [hunk 3]
```diff
@@ -285,7 +285,7 @@ const createConfig = (
   }
 
   // Optimize the bundle in production mode
-  if (!devServer && optimize) {
```
Comment (maxammann) [inline web/tools/webpack.config.ts:288]: This changes the meaning of the code I suppose.

**Review by maxammann — APPROVED** [review]

File: web/src/components/Tooltip.tsx (hunk 2 above)
Comment (klinzo) [reply web/src/components/Tooltip.tsx:222]: It is guaranteed by typescript that there are only valid keys used, so they should not be necessary.

File: web/tools/webpack.config.ts (hunk 3 above)
Comment (klinzo) [reply web/tools/webpack.config.ts:288]: optimize is just a namespace/import from webpack, I don't really see a reason why we need to check for it.

File: build-configs/tsconfig.json (hunk 1 above)
Comment (klinzo) [reply build-configs/tsconfig.json:2]: To enable these rules with typescript support I had to set the parserOptions in .eslintrc.js to a ts config. To allow linting from the root dir as well as from the different workspaces the same name is needed. At least I don't know how to allow that otherwise.

**Review by SteffiStuffel — APPROVED** [review]

**Comment by klinzo** [comment]
@maxammann could you have a look again please on my comments?

File: web/src/components/Tooltip.tsx (hunk 2 above)
Comment (maxammann) [reply web/src/components/Tooltip.tsx:222]: spaceCheckers doesnt need to have every key present. The if check can be true if you change the config above.
